src/yb/yql/pgwrapper/pg_wrapper.cc

```diff
@@ -349,14 +357,37 @@ Status PgWrapper::Start() {
   }
 
   argv.push_back("-c");
-  // TODO: we should probably load the metrics library in a different way once we let
-  // users change the shared_preload_libraries conf parameter.
+
+  // Gather the default extensions:
+  vector<string> metricsLibs;
   if (FLAGS_pg_stat_statements_enabled) {
-    argv.push_back("shared_preload_libraries=pg_stat_statements,yb_pg_metrics,pgaudit,"
-      "pg_hint_plan");
+    metricsLibs.push_back("pg_stat_statements");
+    metricsLibs.push_back("yb_pg_metrics");
```

Comment:
It seems this is the difference between the two branches.
You can lift the statements from else block up. And leave only this line in if block (to reduce code dup).

Reply:
Makes sense, done in https://github.com/yugabyte/yugabyte-db/pull/9576/commits/dc53f3c3b497f605e6dd7a97a2a03b12ed581d37 but not sure if the order is significant so left it with the if branch in the middle.

Reply:
I agree to keep the order.